bitstreamformats.md

```diff
@@ -0,0 +1,96 @@
+# Bitstream Formats Endpoints
+[Back to the list of all defined endpoints](endpoints.md)
+
+## Main Endpoint
+**GET /api/core/bitstreamformats**   
+
+Provide access to the bitstream formats defined in the registry (DBMS based). It returns the list of existent metadata fields.
+
+Example: <https://dspace7.4science.cloud/dspace-spring-rest/#https://dspace7.4science.cloud/dspace-spring-rest/api/core/bitstreamformats>
+
+**POST /api/core/bitstreamformats**   
+
+Create a new bitstream format in the registry (DBMS based). Requires an admin account
+
+The JSON should be similar to:
+```json
+{
+  "shortDescription": "XML",
+  "description": "Extensible Markup Language",
+  "mimetype": "text/xml",
+  "supportLevel": 1,
```

Comment:
If we are going to return the `supportLevel` as the stored integer (0, 1 or 2), then we likely should describe in the Contract which values are valid and what each value means.

There's only three valid values, 0, 1 or 2.  The index corresponds to the text listed here: https://github.com/DSpace/DSpace/blob/master/dspace-api/src/main/java/org/dspace/content/BitstreamFormatServiceImpl.java#L52-L53

The other option here would be to expect the supportLevelText value in these JSON representations.  So, we could think about replacing this line with `"supportLevel": "KNOWN"`

To be honest, I don't have a strong opinion here. But, we likely don't want to hardcode the same set of constants on the client side (Angular).  So, that may imply we should be returning the string values here instead of integers.

Reply:
Thanks for the useful suggestion @tdonohue 
I did copy that part from the current implementation at https://dspace7.4science.cloud/dspace-spring-rest/#https://dspace7.4science.cloud/dspace-spring-rest/api/core/bitstreamformats, but I agree that it's much better to return  `"supportLevel": "KNOWN"`
I will update the contract accordingly